libdnf/module/module_item.cpp

```diff
@@ -122,6 +122,17 @@ std::string ModuleItem::get_name_stream_context() const {
 }
 
 
+std::string ModuleItem::get_name_stream_staticcontext_arch() const {
+    return libdnf::utils::sformat(
+        "{}:{}:{}:{}",
+        libdnf::utils::string::c_to_str(modulemd_module_stream_get_module_name(md_stream)),
+        libdnf::utils::string::c_to_str(modulemd_module_stream_get_stream_name(md_stream)),
+        computed_static_context.empty() ? libdnf::utils::string::c_to_str(modulemd_module_stream_get_context(md_stream))
+                                        : computed_static_context,
+        libdnf::utils::string::c_to_str(modulemd_module_stream_get_arch(md_stream)));
+}
```

Comment:
If you would not care about performance optimization you could shorter the code using `get_name_stream_staticcontext()` call. I guess that the performance difference will be minimal.

Reply:
Well, there are many methods like this. If we used this approach everywhere, it would result in multiple calls in most of them. And if it was used only sometimes, it would be inconsistent. I don't think it would really improve readability, but of course, it would be shorter. So, I don't know - if you think it would be better, I can change it.